src/software/physics/euclidean_to_wheel.cpp

```diff
@@ -76,3 +80,109 @@ EuclideanSpace_t EuclideanToWheel::getEuclideanVelocity(
 
     return euclidean_velocity;
 }
+
+WheelSpace_t EuclideanToWheel::rampWheelVelocity(
+    const WheelSpace_t& current_wheel_velocity,
+    const EuclideanSpace_t& target_euclidean_velocity, const double& time_to_ramp)
+{
+    double allowed_acceleration =
+        static_cast<double>(robot_constants.robot_max_acceleration_m_per_s_2);
+    double max_allowable_wheel_velocity =
+        static_cast<double>(robot_constants.robot_max_speed_m_per_s);
+
+    // ramp wheel velocity
+    WheelSpace_t ramp_wheel_velocity;
+
+    // calculate max allowable wheel velocity delta using dv = a*t
+    auto allowable_delta_wheel_velocity = allowed_acceleration * time_to_ramp;
+
+    // convert euclidean to wheel velocity
+    WheelSpace_t target_wheel_velocity = getWheelVelocity(target_euclidean_velocity);
+
+    // Ramp wheel velocity vector
+    // Step 1: Find absolute max velocity delta
+    auto delta_target_wheel_velocity = target_wheel_velocity - current_wheel_velocity;
+    auto max_delta_target_wheel_velocity =
+        delta_target_wheel_velocity.cwiseAbs().maxCoeff();
+
+    // Step 2: Compare max delta velocity against the calculated maximum
+    if (max_delta_target_wheel_velocity > allowable_delta_wheel_velocity)
+    {
+        // Step 3: If larger, scale down to allowable max
+        ramp_wheel_velocity =
+            (delta_target_wheel_velocity / max_delta_target_wheel_velocity) *
+                allowable_delta_wheel_velocity +
+            current_wheel_velocity;
+    }
+    else
+    {
+        // If smaller, go straight to target
+        ramp_wheel_velocity = target_wheel_velocity;
+    }
+
+    // find absolute max wheel velocity
+    auto max_ramp_wheel_velocity = ramp_wheel_velocity.cwiseAbs().maxCoeff();
+
+    // compare against max wheel velocity
+    if (max_ramp_wheel_velocity > max_allowable_wheel_velocity)
+    {
+        // if larger, scale down to max
+        ramp_wheel_velocity = (ramp_wheel_velocity / max_ramp_wheel_velocity) *
+                              max_allowable_wheel_velocity;
+    }
+
+    return ramp_wheel_velocity;
+}
+
+std::unique_ptr<TbotsProto::DirectControlPrimitive> EuclideanToWheel::rampWheelVelocity(
+    const std::pair<Vector, AngularVelocity> current_velocity,
+    TbotsProto::DirectControlPrimitive& target_velocity_primitive,
+    const double& time_to_ramp)
+{
+    EuclideanSpace_t target_euclidean_velocity = {};
```

Comment:
Since there's a chance that we might not fill this vector, this could be a dangerous operator (speaking from previous experience :sweat_smile:). Mainly because `Eigen::Vector3d` doesn't have a default constructor that initializes its fields, meaning that this Vector could have any random value in it. As such, we should always try to initialize these vectors/matrices to 0 using
```suggestion
    EuclideanSpace_t target_euclidean_velocity = EuclideanSpace_t::Zero();
```